tests/Unit/MicroplateTest.php

```diff
@@ -0,0 +1,324 @@
+<?php declare(strict_types=1);
+
+namespace Mll\Microplate\Tests\Unit;
+
+use Mll\Microplate\Coordinate;
+use Mll\Microplate\CoordinateSystem12Well;
+use Mll\Microplate\CoordinateSystem96Well;
+use Mll\Microplate\Enums\FlowDirection;
+use Mll\Microplate\Exceptions\MicroplateIsFullException;
+use Mll\Microplate\Exceptions\WellNotEmptyException;
+use Mll\Microplate\Microplate;
+use PHPUnit\Framework;
+
+class MicroplateTest extends Framework\TestCase
+{
+    public function testCanAddAndRetriedWellBasedOnCoordinateSystem(): void
+    {
+        $microplate = new Microplate(new CoordinateSystem96Well());
+
+        $microplateCoordinate1 = new Coordinate('A', 2);
+        $microplateCoordinate2 = new Coordinate('A', 3);
+
+        $wellContent1 = 'foo';
+        $microplate->addWell($microplateCoordinate1, $wellContent1);
+
+        $wellContent2 = 'bar';
+        $microplate->addWell($microplateCoordinate2, $wellContent2);
+
+        self::assertEquals($wellContent1, $microplate->wells[$microplateCoordinate1->toString()]);
```

Comment:
```suggestion
        self::assertEquals($wellContent1, $microplate->well($microplateCoordinate1));
```

That way, the implementation details can be hidden